processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/NodeDetailsDeriver.java

```diff
@@ -80,11 +82,12 @@ public NodeDetails processOneToOne(String tenantId, Span item) throws RetryAttem
         // TODO: How to calculate actual - i.e. would need to know child times???
         nd.setActual(item.getDuration());
 
-        //ct.setIpAddress(item.getAnnotations().get(0).getEndpoint().getIpv4());
-
         // TODO: ADD SERVICE NAME AS PROPERTY?
 
+        List<Property> spanProperties = item.properties();
         nd.setTimestamp(item.getTimestamp() / 1000);
+        nd.getProperties().addAll(spanProperties);
+        nd.setHostName(Span.ipv4Address(spanProperties));
```

Comment:
As with CompletionTime - can you add a hostAddress property and set this instead of hostName.